### PR TITLE
feat(DCS): restore dcs by backup id

### PR DESCRIPTION
**Pull request by 346010762**
**What this PR does / why we need it**:
restore dcs by backup id
**Special notes for your reviewer**:

**Release note**:

```release-note
restore dcs by backup id
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dcs/ TESTARGS='-run TestAccDcsRestore_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs/ -v -run TestAccDcsRestore_basic -timeout 360m -parallel 4
=== RUN   TestAccDcsRestore_basic
=== PAUSE TestAccDcsRestore_basic
=== CONT  TestAccDcsRestore_basic
--- PASS: TestAccDcsRestore_basic (196.19s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs       196.230s

```

### CODE REVIEW

File: docs/resources/dcs_restore.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+* `create` - Default is 30 minutes.
+
+## Import
+
+The restore record can be imported using the DCS instance ID and the restore record ID separated by a slash, e.g.
+
+```bash
+$ terraform import huaweicloud_dcs_restore.test <instance_id>/<id>
+```
```
Comment (houpeng80) [inline docs/resources/dcs_restore.md]: import should not be supported

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_restore.go [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+	d.SetId(restoreId.(string))
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"waiting", "restoring"},
+		Target:       []string{"succeed"},
+		Refresh:      restoreRecordRefreshFunc(instanceID, d.Id(), restoreDcsCreateClient),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        2 * time.Second,
+		PollInterval: 2 * time.Second,
```
Comment (houpeng80) [inline huaweicloud/services/dcs/resource_huaweicloud_dcs_restore.go]:
```suggestion
		Delay:         10 * time.Second,
		PollInterval: 10 * time.Second,
```

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_restore.go [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+			return nil, fmt.Errorf("error retrieving restore records: %s", err)
+		}
+
+		getRestoreRespBody, err := utils.FlattenResponse(getRestoreResp)
+		if err != nil {
+			return nil, err
+		}
+
+		restoreRecords := utils.PathSearch("restore_record_response", getRestoreRespBody, make([]interface{}, 0)).([]interface{})
+		totalNum := utils.PathSearch("total_num", getRestoreRespBody, 0)
```
Comment (houpeng80) [inline huaweicloud/services/dcs/resource_huaweicloud_dcs_restore.go]:
```suggestion
		totalNum := utils.PathSearch("total_num", getRestoreRespBody, float64(0))
```

File: docs/resources/dcs_restore.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this creates a new resource.
+
+* `instance_id` - (Required, String, ForceNew) Specifies the ID of the DCS instance.
+  Changing this creates a new resource.
+
+* `backup_id` - (Required, String, ForceNew) Specifies the backup ID of the DCS instance.
+  Changing this creates a new resource.
+
+* `description` - (Optional, String, ForceNew) Specifies the description about restoring the DCS instance.
```
Comment (houpeng80) [inline docs/resources/dcs_restore.md]:
```suggestion
* `description` - (Optional, String, ForceNew) Specifies the description of the DCS instance restoration.
```

File: docs/resources/dcs_restore.md (hunk 1 above)
Comment (346010762) [reply docs/resources/dcs_restore.md]: deleted

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_restore.go (hunk 2 above)
Comment (346010762) [reply huaweicloud/services/dcs/resource_huaweicloud_dcs_restore.go]: done

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_restore.go (hunk 3 above)
Comment (346010762) [reply huaweicloud/services/dcs/resource_huaweicloud_dcs_restore.go]: done

File: docs/resources/dcs_restore.md (hunk 4 above)
Comment (346010762) [reply docs/resources/dcs_restore.md]: done

File: docs/resources/dcs_restore.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+---
+subcategory: "huaweicloud_dcs_restore"
```
Comment (Jason-Zhang9309) [inline docs/resources/dcs_restore.md]:
```suggestion
subcategory: "huaweicloud_dcs_isntance_restore"
```

File: docs/resources/dcs_restore.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+---
+
+# huaweicloud_dcs_restore
+
+Manages a DCS restoration record resource within HuaweiCloud.
```
Comment (Jason-Zhang9309) [inline docs/resources/dcs_restore.md]:
```suggestion
Use this resource to restore a DCS instance with a backup within HuaweiCloud.
```

File: docs/resources/dcs_restore.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this creates a new resource.
+
+* `instance_id` - (Required, String, ForceNew) Specifies the ID of the DCS instance.
```
Comment (Jason-Zhang9309) [inline docs/resources/dcs_restore.md]:
```suggestion
* `instance_id` - (Required, String, ForceNew) Specifies the ID of the DCS instance to be restored.
```

File: docs/resources/dcs_restore.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  Changing this creates a new resource.
+
+* `backup_id` - (Required, String, ForceNew) Specifies the backup ID of the DCS instance.
```
Comment (Jason-Zhang9309) [inline docs/resources/dcs_restore.md]:
```suggestion
* `backup_id` - (Required, String, ForceNew) Specifies the backup ID used to restore the DCS instance.
```

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_restore.go [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+
+	restoreDcsCreateRespBody, err := utils.FlattenResponse(restoreDcsCreateResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	restoreId, err := jmespath.Search("restore_id", restoreDcsCreateRespBody)
+	if err != nil {
+		return diag.Errorf("error restoring the instance (%s): %s", instanceID, err)
+	}
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dcs/resource_huaweicloud_dcs_restore.go]:
```suggestion
	restoreId := utils.PathSerach("restore_id", restoreDcsCreateRespBody, "")
	if restoreId == "" {
		return diag.Errorf("unable to find the restore_id of the instance (%s)", instanceID)
	}
```

File: docs/resources/dcs_restore.md (hunk 5 above)
Comment (346010762) [reply docs/resources/dcs_restore.md]: done

File: docs/resources/dcs_restore.md (hunk 6 above)
Comment (346010762) [reply docs/resources/dcs_restore.md]: done

File: docs/resources/dcs_restore.md (hunk 7 above)
Comment (346010762) [reply docs/resources/dcs_restore.md]: done

File: docs/resources/dcs_restore.md (hunk 8 above)
Comment (346010762) [reply docs/resources/dcs_restore.md]: done

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_restore.go (hunk 9 above)
Comment (346010762) [reply huaweicloud/services/dcs/resource_huaweicloud_dcs_restore.go]: done

**Pull request merged**